src/Utils/ObjectMixin.php

```diff
@@ -317,6 +306,45 @@ public static function getMagicMethods($class)
 
 
 	/**
+	 * Returns array of magic properties defined by annotation @property.
+	 * @return array
+	 */
+	public static function getMagicProperties($class)
+	{
+		$props = & self::$properties[$class];
+		if ($props !== NULL) {
+			return $props;
+		}
+
+		$rc = new \ReflectionClass($class);
+		preg_match_all('~^
+			[ \t*]*  @property(|-read|-write)  [ \t]+
+			(?: [^\s]+  [ \t]+ )?
+			\$ (\w+)
+		()~mx', (string) $rc->getDocComment(), $matches, PREG_SET_ORDER);
+
+		$props = [];
+		foreach ($matches as list(, $type, $name)) {
+			$uname = ucfirst($name);
+			$write = (!$type || $type === '-write')
+				&& $rc->hasMethod($nm = 'set' . $uname)
+				&& ($rm = $rc->getMethod($nm)) && $rm->getName() === $nm && !$rm->isPrivate() && !$rm->isStatic();
+			$read = (!$type || $type === '-read')
+				&& ($rc->hasMethod($nm = 'get' . $uname) || $rc->hasMethod($nm = 'is' . $uname))
+				&& ($rm = $rc->getMethod($nm)) && $rm->getName() === $nm && !$rm->isPrivate() && !$rm->isStatic();
+			if ($read || $write) {
+				$props[$name] = [$read ? $nm : FALSE, $write, $read ? $rm->returnsReference() : NULL];
```

Comment:
This first only will almost always match (90+ % by guess), the second one will handle the most of the remaining cases (7 % by guess). Getters returning references (e.g. `& getName()` or `& isValid()`) accessed as properties  are extremely rare.